### ISSUE TITLE
CNN model Experament

### ISSUE THREAD

**Issue by kairos03** [KIST-Iceberg/Iceberg#14]
## 기준 모델
LEARNING_RATE = 1e-3
TOTAL_EPOCH = 150
BATCH_SIZE = 500
DROPOUT_RATE = 0.5

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

## Batch Normalization 위치
 - [x] conv 앞 **Best**
 - [x] ~conv 와 activation 사이~

## Regularization
 - [x] ~사용 안함~
 - [x] ~L2~ 
 - [x] Dropout **Best**

## model
 - [x] simple cnn
 - [x] ~semi-supervised learning (https://pure.qub.ac.uk/portal/files/135794487/3105053.pdf)~ **NOT WORK**
 - [x] 모델변경
  - 4 conv (3x3 kernel, 1x1 strides, 64, 128 128, 64 filter num)
  - 4 max_pool (3x3 k_size, 2x2 strides)
  - 1 dense (512)

**Comment by kairos03** [comment]
https://www.kaggle.com/c/statoil-iceberg-classifier-challenge/discussion/46686
Data Argumentation시 Flip을 사용하면 Inc_angle을 무시하고 학습이 가능하고, 앙상블 및 다른 모델과 결합하여 사용하는 것을 추천 
Flip을 사용하지 않은 데이터셋은 inc_angle을 포함하여 Single model일 경우 더 나은 성능을 보임. 
 #7 #8 #9

**Comment by kairos03** [comment]
# EXP No. 1

## 모델
LEARNING_RATE = 1e-3
TOTAL_EPOCH = **260**
BATCH_SIZE = 500
DROPOUT_RATE = 0.5

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**L2 Regularization 적용**

![image](https://user-images.githubusercontent.com/6357456/34567306-08e7c724-f162-11e7-86c1-5b568a3c3ce2.png)

## 결론
주황색 부분이며 일정 부분 학습 후 학습이 더 이상 진행 되지 않는것으로 보임.
Overfitting 되는것을 방지하는것은 효과있음.

**Comment by kairos03** [comment]
# EXP No. 2

## 모델
LEARNING_RATE = 1e-3
TOTAL_EPOCH = 150
BATCH_SIZE = 500
**DROPOUT_RATE = 0.9**

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**L2 Regularization 적용**

![image](https://user-images.githubusercontent.com/6357456/34567596-275453ca-f163-11e7-9ce4-36d802936835.png)

## 결론
진한 파란색 부분이며, Dropout 과  Regularization 은 상관성이 없어보임.

**Comment by kairos03** [comment]
# EXP No. 2

## 모델
LEARNING_RATE = 1e-3
**TOTAL_EPOCH = 120**
BATCH_SIZE = 500
DROPOUT_RATE = 0.5
**REGULARIZATION_BETA = 0.001**

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**L2 Regularization 적용**

![image](https://user-images.githubusercontent.com/6357456/34568089-fbc70ebc-f164-11e7-8aed-ec9ffcfff62e.png)


## 결론
Train: 자주색, Test: 초록색.
Regularization beta 값을 낮추니 이전보다는 loss가 적게 나옴.

**Comment by kairos03** [comment]
# EXP No. 3

## 모델
LEARNING_RATE = 1e-3
**TOTAL_EPOCH = 300**
BATCH_SIZE = 500
DROPOUT_RATE = 0.2
**REGULARIZATION_BETA = 5e-4**

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**L2 Regularization 적용**
**Lee Filter로 변경**

![image](https://user-images.githubusercontent.com/6357456/34572807-a177619c-f172-11e7-8645-04a95b2348aa.png)

## 결론
Train: 회색, Test: 주황색.
Lee filter 적용 시 Accuracy에 큰 변화가 있음.
http://cradpdf.drdc-rddc.gc.ca/PDFS/unc196/p800002_A1b.pdf

**Comment by kairos03** [comment]
# EXP No. 4

## 모델
LEARNING_RATE = 1e-3
**TOTAL_EPOCH = 500**
BATCH_SIZE = 600
DROPOUT_RATE = 0.2
**REGULARIZATION_BETA = 1e-3**

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,18 | stride:1, filter: 7*7 |
| maxpool1 |  15, 15, 18 | stride:5 |
| conv2 |  15,15,36 | stride:1, filter: 5*5 |
| maxpool2 |  3,3,36 | stride:5 | 
| flatten | 324 | |
| add_angle | 325 | |
| dense | 120 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**L2 Regularization 적용**
**Lee Filter로 변경**

![image](https://user-images.githubusercontent.com/6357456/34611607-6fafa200-f226-11e7-8211-8eeaac42c3c1.png)


## 결론
Train: 하늘색, Test: 자주색.
Lee filter 적용 시 Accuracy에 큰 변화가 있음. 
LB : 0.7785

**Comment by kairos03** [comment]
# EXP No. 5

## 모델
LEARNING_RATE = 1e-3
**TOTAL_EPOCH = 500**
BATCH_SIZE = 1000
DROPOUT_RATE = 0.2

| 레이어 | shape | detail |
| --- | --- | --- |
| input | 75, 75, 3 | |
| conv1 |  75,75,64 | stride:1, filter: 3*3 |
| maxpool1 |  38, 38, 64 | stride:2 |
| conv2 |  38,38,128 | stride:1, filter: 3*3 |
| maxpool2 |  19,19,128 | stride:2 | 
| conv3 |  19,19,128 | stride:1, filter: 3*3 |
| maxpool3 |  10, 10, 128 | stride:2 |
| conv1 |  10,10,64 | stride:1, filter: 3*3 |
| maxpool1 |  5, 5, 64 | stride:2 |
| flatten | 1600 | |
| add_angle | 1601 | |
| dense | 512 | |
| output | 2 | |

**Batch Normalization 레이어 앞**
**Lee Filter로 변경**
**모델변경**
**Optimizer: SGD**


[사진추가예정]

## 결론
Train: , Test: .
모델이 너무 작은것 같아 조금 크게 확장
optimizer ADAM을 사용하면 중간중간 값이 크게 튐으로 SGD사용 >> 시간 오래걸림
